corehq/apps/userreports/tasks.py

```diff
@@ -63,7 +67,14 @@ def _get_redis_key_for_config(config):
     if not is_static:
         client.delete(redis_key)
         config.meta.build.finished = True
-        config.save()
+        try:
+            config.save()
+        except ResourceConflict:
+            current_config = DataSourceConfiguration.get(config._id)
+            # check that a new build has not yet started
+            if config.meta.build.initiated == current_config.meta.build.initiated:
```

Comment:
we could also check for `and not meta.build.finished` to prevent a redundant save (though I don't actually have any scenario in my head of when it could get into that state so maybe not even a good idea)